### DISCUSSION TITLE
NS entries download failed: javax.net.ssl.SSLHandshakeException: Java.security.cert.CertPathValidatorException. Trust anchor for certification path not found

### DISCUSSION THREAD

**Discussion by giuseppe-Marfisi** [NightscoutFoundation/xDrip#3342] [Q&A] [answered]
Da qualche giorno il mio tablet che utilizzo come follower non mi fa visualizzare più i valori della glicemia e visualizza il seguente errore:
NS entries download failed: javax.net.ssl.SSLHandshakeException: Java.security.cert.CertPathValidatorException. Trust anchor for certification path not found.
Ho verificato tutte le impostazioni e sono identiche agli altri follower che su altri tablet e cellulari funzionano correttamente.
Mi potete aiutare per favore?
P. S. : Sul Tablet, su cui gira Android 7.0, ho installato l'ultima versione di xdrip (ma non cambia nulla anche con precedenti release). Il master é un iPhone 14 con shuggar.

**Comment by Navid200** [comment]
What version of xDrip are you using?
Can you see readings if you access your Nightscout in a browser?

**Reply by giuseppe-Marfisi** [reply]
x-drip version is no.  8da5e5d-2024-02-07. 
I also tried installing previous versions but the result doesn't change

Nightscout works great on the browser.

**Reply by Navid200** [reply]
So, Nightscout shows results as expected in a browser.
But, xDrip does not show readings as a follower of Nightscout?

If yes, can you show a screenshot of the xDrip logs?

**Reply by giuseppe-Marfisi** [reply]
![Screenshot_20240214-224355_104507](https://github.com/NightscoutFoundation/xDrip/assets/29926586/bde5f68f-11bf-47be-b8b6-d3fac69a3369)

**Reply by Navid200** [reply]
Are you sure you have entered the correct Nightscout hostname, token and/or API_SECRET in your Nightscout follower??

**Reply by giuseppe-Marfisi** [reply]
Yes

**Reply by giuseppe-Marfisi** [reply]
I even copied and pasted it with the other followers and it works

**Reply by Navid200** [reply]
> I even copied and pasted it with the other followers and it works

What works?

**Reply by giuseppe-Marfisi** [reply, marked as answer]
Nightscout's link to other followers.  But nothing on my tablet.  No value arrives

**Reply by giuseppe-Marfisi** [reply]
doing some research on the internet I think I understood that the problem comes from the lack of an installed certificate or from the fact that the one installed is no longer considered reliable.  I haven't yet understood which one and how to replace it with one signed by the Authority that is considered reliable SSL/TLS

**Reply by PhilTU** [reply]
it is possible  Android 7.0 will not support latest security requirements ?

you can see similar issues on the internet like https://community.letsencrypt.org/t/letsencrypt-certificates-fails-on-android-phones-running-android-7-or-older/205686 which appear to relate to a specific type of encryption no longer working.

Also check the time & data is corrent.

**Reply by giuseppe-Marfisi** [reply]
Nessun'altra aiuto? Please!

**Reply by Navid200** [reply]
Where is your Nightscout server?
Is it Heroku?

**Comment by giuseppe-Marfisi** [comment]
Railway
google

Il giorno dom 10 mar 2024 alle ore 22:25 Navid ***@***.***>
ha scritto:

**Reply by Navid200** [reply]
If Railways uses Let's encrypt, this could explain why you are experiencing this problem:
https://www.ispmanager.com/news/let-s-encrypt-reduces-its-support-for-android-7

**Reply by giuseppe-Marfisi** [reply]
I read what was written on this link and installed Firefox, but nothing changed. Unfortunately, xdrip doesn't even work using google's server

**Reply by Navid200** [reply]
That's an explanation what the cause of the problem is assuming Railway uses Let's encrypt.  It is not a suggested solution.
If railway uses Let's encrypt, you cannot use Android 7.

**Reply by giuseppe-Marfisi** [reply]
In your opinion, how can I solve the problem?

**Reply by Navid200** [reply]
Ask Railway or someone else who may know if they use Let's Encrypt.
If the answer is yes, you need a different tablet with Android 8 or newer.

**Reply by giuseppe-Marfisi** [reply]
I must add that I have another Android 7 tablet which however works correctly. This would make me think that the problem is different. In those days I had installed some streaming apps. Is it possible that some certificates have been damaged?

**Reply by Navid200** [reply]
Then, I'm not sure what's going on.

**Reply by PhilTU** [reply]
Does https://valid-isrgrootx1.letsencrypt.org/ look the same on both tablets?

**Reply by giuseppe-Marfisi** [reply]
![Screenshot_20240311-101018](https://github.com/NightscoutFoundation/xDrip/assets/29926586/e210550e-a061-4492-a077-f09642a0ec70)
Yes.  the attached image is displayed on both tablets.  Bear in mind that the other tablet has version 8 of Android

**Reply by PhilTU** [reply]
ok, good.  maybe I lost track of how many tablets you have :-)

**Reply by giuseppe-Marfisi** [reply]
simply 2. the android 7 one doesn't work;  the one with android 8 works

**Reply by Navid200** [reply]
But, you said you had another Android 7 tablet that worked.  Now, you say the one that works is 8.  
If that is the case, we already went over the cause.
I mean you can talk about this all you want.  But, it's not going to change anything.

**Reply by giuseppe-Marfisi** [reply]
I don't think I ever said I had another Android 7 tablet that works.  Unless this was due to a translation error from Italian into English.  For clarity I reiterate that I have two tablets.  the one with Android 7 doesn't work;  the one with Android 8 works.  I was asking for your help in making changes to the tablet that isn't working and ensuring that it works again soon, allowing me to view my daughter's blood sugar levels.  It's a health issue and that's why I ask you to help me.  If you think the problem cannot be solved, tell me and I thank you anyway.

**Reply by Navid200** [reply]
If the server uses Let's encrypt, Android 7 does will not work any longer after the certificate expires.
You can either switch to a different Nightscout server option that does not use Let's encrypt.  Or, you can switch to using a tablet that has Android 8 or newer.

**Reply by giuseppe-Marfisi** [reply]
Forgive me for my ignorance on the subject.  but the let's Encrypt test was not passed?  I have already posted the screenshot of my tablet where I think it says that the certificate is valid.  or maybe I'm wrong?  Also, when you talk about Nightscout servers that don't use let's Encrypt, do you already have a solution in mind or do I have to look online?  I take advantage of your availability and competence and therefore I thank you from the bottom of my heart

**Reply by Navid200** [reply]
No, I don't know any alternative that does not use Let's encrypt.

I can tell you that Google Cloud uses Let's encrypt.  So, you will have the same problem there.

**Reply by giuseppe-Marfisi** [reply]
yes, in fact, I tried with Google but there was nothing to do there either.  Thanks for your reply.

**Reply by Navid200** [reply]
You misunderstood.
An option for a Nightscout server is Google Cloud Nightscout instead of Railway.  This is information for using Google Cloud Nightscout:
https://navid200.github.io/xDrip/docs/Nightscout/GoogleCloud.html

And I meant that option uses Let's encrypt and as a result Android 7 will have trouble using it.

**Comment by giuseppe-Marfisi** [comment]
I read what was written on this link and installed Firefox, but
nothing changed. Unfortunately, xdrip doesn't even work using google's
server


Il giorno dom 10 mar 2024 alle ore 23:58 Navid ***@***.***>
ha scritto:

**Comment by PhilTU** [comment]
> I don't think I ever said I had another Android 7 tablet that works. Unless this was due to a translation error from Italian into English. For clarity I reiterate that I have two tablets. the one with Android 7 doesn't work; the one with Android 8 works.

You wrote earlier "I must add that I have another Android 7 tablet which however works correctly. " which may have been a simple typo.
